### DISCUSSION TITLE
Uncaught TypeError: $(...).jstree is not a function

### DISCUSSION THREAD

**Discussion by charlieliu2001** [vakata/jstree#2626] [Q&A]
Good day. I'm trying to obtain the nodes that are checked via the checkbox, but it is throwing the error `Uncaught TypeError: $(...).jstree is not a function`.

I made sure to check the posts others have made on this error, but I still wasn't able to resolve it.

I have jquery and jstree imported like so:

```
<script src="/Scripts/jquery-3.2.1.js"></script>
<script src="/Scripts/jstree/jstree.js"></script>
```

The code segment in question is:

```
    $('#jstree')
        // listen for event
        .on('check_node.jstree', function (e, data) {
            var checked_ids = $('#jstree').jstree(true).get_selected();
        })
        // create the instance
        .jstree();
```

I have also tried `$("#jstree").jstree().get_checked()`

I'm not sure how to fix this because everything else is working fine.

My tree is defined as follows. Weird because it used .jstree() just fine without error.

```
    $('#jstree').jstree({
        'core': {
            'themes': {
                'responsive': false,
                'multiple': false,
            },
            'data':
                @Html.Raw(Newtonsoft.Json.JsonConvert.SerializeObject(tree))
        },
        "types": {
            "default": {
                "icon": "glyphicon glyphicon-folder-open"
            }
        },
        "ui": {
            "select_limit": 1,
        },
        "plugins": ["wholerow", "types", "checkbox", "ui", "crrm", "sort"],
        "checkbox": {
            "three_state": false,
            "real_checkboxes": true,
            "whole_node" : false,  // to avoid checking the box just clicking the node 
            "tie_selection": false, // for checking without selecting and selecting without checking
            "cascade": 'down',
            "keep_selected_style" : 'false'
        }
    });
```

Any help is greatly appreciated!

My final goal is only having one checked at a time [like so](https://stackoverflow.com/questions/64692547/jstree-only-allow-to-check-one-node-each-time-in-all-the-jstree).

Error:

![image](https://user-images.githubusercontent.com/43513073/167924282-c753f3a0-ce55-4042-8d5d-8351a636b974.png)

Update: Issue fixed - was due to layout page having conflicts

**Comment by Stefano-Buzzoni** [comment]
Hi, could you give me some more explanations? I have the same problem and I just can't fix it, I've tried so many solutions but I can't manage to get it works. Using ASP NET Core on MVC application

**Reply by charlieliu2001** [reply]
In my case, I had a _Layout.cshtml that I was using which included a lower version of jquery, which conflicted with my specifying the jquery version again in my view via `<script src="/Scripts/jquery-3.2.1.js"></script>`. When I fixed the jquery version in my Layout page I managed to solve the problem. Hope this helps!

**Comment by Stefano-Buzzoni** [comment]
Il giorno mar 24 mag 2022 alle 13:37 charlieliu2001 <
***@***.***> ha scritto:

> In my case, I had a _Layout.cshtml that I was using which included a lower
> version of jquery, which conflicted with my specifying the jquery version
> again in my view via <script src="/Scripts/jquery-3.2.1.js"></script>.
> When I fixed the jquery in my Layout page I managed to solve the problem.
> Hope this helps!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vakata/jstree/discussions/2626#discussioncomment-2811340>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP4L33LU7CIYZYGH25ONEULVLS5RJANCNFSM5VV3F4JA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
Hi, thanks for your reply. After reading more carefully your statement I
figured it out my self but thanks anyway for your reply :-)
